File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
function isPimaryNumber(num) {
    if (num === '1' || num === '') return false;
    
    for (let i = 2; i <= Math.sqrt(+num); i++) {
        if (+num % i === 0) return false;
    }
    return true;
}

function solution(n, k) {
    const result = [];
    const number = n.toString(k);
    const list = number.split('0');

    for (let i = 0; i < list.length; i++) {
        if (isPimaryNumber(list[i])) {
            result.push(list[i]);
        }
    }
    return result.length;
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def to_base_k(num, base):
    digit = '0123456789'
    temp = ''
    while num > 0:
        temp = digit[num % base] + temp
        num //= base
    
    return temp

def is_prime(num):
    if num < 2: return False

    for i in range(2, int(num ** 0.5) + 1):
        if num % i == 0:
            return False
    
    return True
    
def solution(n, k):
    result = 0
    k_base = to_base_k(n, k).split('0')
    
    for num in k_base:
        if num and is_prime(int(num)):
            result += 1
    
    return result
    
    